### ISSUE TITLE
MISC: Issues with Typescript support

### ISSUE THREAD

**Issue by G4mingJon4s** [bitburner-official/bitburner-src#1488]
Since the introduction of `.jsx`, `.ts` and `.tsx` file in #1216, players can use typescript. This is honestly one of the greatest changes ever, thank you very much  @catloversg for the great work!

There are some small things I found when playing around. I hope it is clear what is working and not by my comments.
I have two files, both `.ts`: `test.ts` and `lib.ts` on the home server, no complicated folder structure nothing.
```ts
// lib.ts
// This function and its signature is exported correctly
export const sum = (a: number, b: number): number => a + b;

// Type extensions from test.ts are also included here even without importing
const received = globalThis.myValue
```

```ts
// test.ts
import { sum } from "lib"

enum MyTest {
  a = 1,
  b = 2,
  c = 3
}

declare global {
  namespace globalThis {
    var myValue: number;
  }
}

export async function main(ns: NS) {
  // Basic types from the game work
  const myServer: Server = {
    hostname: "a",
    ip: "0.0.0.0",
    sshPortOpen: false,
    httpPortOpen: false,
    smtpPortOpen: false,
    sqlPortOpen: false,
    ftpPortOpen: false,
    hasAdminRights: false,
    cpuCores: 1,
    isConnectedTo: false,
    ramUsed: 0,
    maxRam: 1024,
    organizationName: "a",
    purchasedByPlayer: false
  };
  const a: CorpEmployeePosition = "Operations";

  try {
    // RAM allocation as before
    await ns["weaken"]("a");
    await ns.grow("a");
  } catch { }

  // Type extension not only works here, but gets carried over to lib.ts
  globalThis.myValue = 5;
  const value = globalThis.myValue;

  // Enums work
  ns.tprint(MyTest.a);

  // Type signatures are now properly ported over
  const c = sum(1, (() => 1 + 2)());

  // But x is not of type 'Task | null'
  const x = ns.singularity.getCurrentWork();
}

// Implicit any is allowed? This should be an error, not a warning?
function greet(name) {
  console.log("Hello " + name.toUpperCase());
}

// This shouldn't be allowed when using TS, I think
const a = 5 + "10";

// Why does this not throw errors?
let name: string;
name = null;
name = undefined;

interface User {
  name: string;
  age: number;
}
const alice: User = { name: "Alice", age: 99, lastname: "this does give an error! yay!" };
```

To recap, the problems I encountered are:
- API functions that might return null are not typed that way.
- Not typing parameters is allowed, it should only be typed as any explicitly though
- Adding a number and a string should result in a compile error, but it doesn't
- You can assign any value null or undefined and it doesn't give an error

I'm not 100% sure, but I think most if not all of these come from the fact that strict mode is disabled.
